pkg/store/postings_codec.go

```diff
@@ -138,41 +140,215 @@ func diffVarintSnappyStreamedDecode(input []byte, disablePooling bool) (closeabl
 }
 
 type streamedDiffVarintPostings struct {
-	cur storage.SeriesRef
+	curSeries storage.SeriesRef
 
-	sr  io.ByteReader
-	err error
+	err               error
+	input, buf        []byte
+	maximumDecodedLen int
+
+	db *encoding.Decbuf
+
+	readSnappyIdentifier bool
+	disablePooling       bool
 }
 
-func newStreamedDiffVarintPostings(input []byte, disablePooling bool) (closeablePostings, error) {
-	if disablePooling {
-		return &streamedDiffVarintPostings{sr: s2.NewReader(bytes.NewBuffer(input))}, nil
+const (
+	chunkTypeCompressedData   = 0x00
+	chunkTypeUncompressedData = 0x01
+	chunkTypeStreamIdentifier = 0xff
+	chunkTypePadding          = 0xfe
+	checksumSize              = 4
```

Comment:
https://github.com/google/snappy/blob/main/framing_format.txt the format hasn't changed in 10 years and it will probably never change. There are other formats based on snappy like s2 but it's very unlikely that snappy will change.